### DISCUSSION TITLE
Is it possible to use context in non-component enviroment?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#376] [Q&A]
I found context is a greate concept and tool for app. Is there some way to use context in non-component environment? such as in synchronous constructor or asynchronous action function?

**Comment by ryansolid** [comment]
It's hard since it's the reactive graph that governs context. I needed to do some hacking with WebComponents to make it work with potential async. Generally speaking with synchronous execution we can support most patterns. Solid's components are functions so if you wanted to do classes you'd need to wrap them, but with async you need to trigger a change in the synchronously created graph to create new reactive primitives, or they will be their own root and you will have to take care of disposal yourself. There are some helpers to insert these isolated parts back in. You can pass an owner to a reactive root to bring it back into the fold. But this is a bit more on the complicated side and I'd probably need to help a bit, as I developed this for my purposes and I'm not sure I've considered all the scenarios.
